l1-contracts/contracts/bridge/ntv/L1NativeTokenVault.sol

```diff
@@ -190,7 +191,11 @@ contract L1NativeTokenVault is IL1NativeTokenVault, IL1AssetHandler, NativeToken
             }
             require(callSuccess, "NTV: claimFailedDeposit failed, no funds or cannot transfer to receiver");
         } else {
-            IERC20(l1Token).safeTransfer(_depositSender, _amount);
+            if (originChainId[_assetId] != block.chainid) {
```

Comment:
The `originChainId` mapping worked in L2AR in such cases easily, since for it the default value in the mapping (0) means that the token was bridged and it is indeed a correct default value. For L1 AR this is not the case.

For instance, here for not-yet-registered tokens we would call `bridgeMint` function. Most certainly the tokens that we ever bridged before do not have this method. I think that an even more robust approach is to create a method called `getOriginChainId` which would:

- If it is set, return whatever is yet
- If it is not set, return L1_CHAIN_ID

The downside is that for tokens that were not yet registered this method would return "L1". Maybe then it makes sense to separate it into "safe" and "unsafe" version.

Reply:
@Raid5594  I think this is the only real issue